src/getters/get-summary.ts

```diff
@@ -0,0 +1,39 @@
+import { truncate } from 'lodash';
+import * as Unist from 'unist';
+
+const { select } = require('unist-util-select');
+const toString = require('mdast-util-to-string');
+
+import { Frontmatter } from '../frontmatter';
+
+export interface IGetSummaryOpts {
+  truncate?: number;
+}
+
+// Priority: yaml summary, then first paragraph
+export const getSummary = (data?: Unist.Node, opts: IGetSummaryOpts = {}) => {
+  let summary: string | void | undefined;
+
+  if (data) {
+    try {
+      const frontmatter = new Frontmatter(data, true);
```

Comment:
```suggestion
      const frontmatter = new Frontmatter(data);
```
You perform a read-only operation here, so no need to clone AST.

Reply:
`true` = mutate = do not clone.

Reply:
🤦‍♂️
true